internal/core/software_renderer.rs

```diff
@@ -1182,6 +1183,24 @@ impl<'a, T: ProcessScene> crate::item_rendering::ItemRenderer for SceneBuilder<'
                 Color::default()
             };
 
+            let border_alpha = border_color.alpha();
+            let mut border_color = PremultipliedRgbaColor::from(border_color);
+            let color = PremultipliedRgbaColor::from(color);
+            if border_alpha == 0 {
+                border = LogicalLength::new(0 as _);
+            } else if border_alpha < 255 {
+                // blend the border color onto the background
+                let b = border_color;
+                let b_alpha_16 = b.alpha as u16;
+                border_color = PremultipliedRgbaColor {
+                    red: ((color.red as u16 * (255 - b_alpha_16)) / 255) as u8 + b.red as u8,
+                    green: ((color.green as u16 * (255 - b_alpha_16)) / 255) as u8 + b.green as u8,
+                    blue: ((color.blue as u16 * (255 - b_alpha_16)) / 255) as u8 + b.blue as u8,
+                    alpha: (color.alpha as u16 + b_alpha_16
```

Comment:
I needed to read this one a few times as it "multiplies out". Looks correct to me though.

Reply:
I added a comment with the maths behind these formulas